usr/share/rear/conf/default.conf

```diff
@@ -3289,37 +3289,44 @@ SIMPLIFY_TEAMING=no
 
 ####
 # Serial console support for the ReaR rescue/recovery system:
-# By default serial consoles get enabled if serial devices are found and
-# then matching kernel command line parameters like 'console=ttyS0,9600 console=ttyS1,9600'
-# are set when booting the rescue/recovery system (see KERNEL_CMDLINE above).
 # IA64 platforms require it and people need it when no VGA console is available.
+# By default there is serial console support in the ReaR recovery system
+# when 'getty' or 'agetty' and 'stty' can be found,
+# otherwise USE_SERIAL_CONSOLE is automatically set to 'no'.
+# When 'getty' or 'agetty' and 'stty' can be found
+# and there is a 'console=...' option in /proc/cmdline,
+# then USE_SERIAL_CONSOLE is automatically set to 'yes'.
```

Comment:
> there is serial console support in the ReaR recovery system
> when 'getty' or 'agetty' and 'stty' can be found,
> otherwise USE_SERIAL_CONSOLE is automatically set to 'no'.

> When 'getty' or 'agetty' and 'stty' can be found
> and there is a 'console=...' option in /proc/cmdline,
> then USE_SERIAL_CONSOLE is automatically set to 'yes'.

What if 'getty' or 'agetty' and 'stty' can be found, but there is no 'console=...' option in /proc/cmdline ?

According to the first rule, USE_SERIAL_CONSOLE is not automatically set to 'no'. But according to the second rule, USE_SERIAL_CONSOLE is not automatically set to 'yes' either. Does USE_SERIAL_CONSOLE remain unset then? And if so, what does it mean to have it unset even after applying this automatism?

Reply:
When 'getty' or 'agetty' and 'stty' can be found,
but there is no 'console=...' option in /proc/cmdline
then 'getty' or 'agetty' and 'stty' get included
in the ReaR recovery system but no serial console
is set up for the ReaR recovery system,
see the code and its comments in
usr/share/rear/prep/GNU/Linux/200_include_serial_console.sh

Reply:
Via
https://github.com/rear/rear/commit/1d07a5a74d7cd1bb2e1d7154378fd9dbe4cbce3b
it now errors out when it cannot find 'getty' or 'agetty' and 'stty'
but USE_SERIAL_CONSOLE is 'true'.

Reply:
According to the comments later `provided USE_SERIAL_CONSOLE is not set to 'no':`, if USE_SERIAL_CONSOLE remains empty, it is equivalent to having the value of 'yes'. Is my understanding correct?

Reply:
When 'getty' or 'agetty' and 'stty' can be found
and there is a 'console=...' option in /proc/cmdline,
then USE_SERIAL_CONSOLE is automatically set to 'yes'.

When 'getty' or 'agetty' and 'stty' can be found
and there is no 'console=...' option in /proc/cmdline,
then USE_SERIAL_CONSOLE remains empty.

When there is no 'console=...' option in /proc/cmdline
no 'console=...' option is cpoied via COPY_KERNEL_PARAMETERS
and the function get_serial_console_devices() results nothing
so no serial console is set up for the recovery system bootloader,
see the comment about "auto-enable" and "auto-aligned" in
prep/GNU/Linux/200_include_serial_console.sh

Reply:
As I understand it, if USE_SERIAL_CONSOLE remains empty, the behavior is as if it is set to 'yes', because all the conditions start with "provided USE_SERIAL_CONSOLE is not set to 'no'". Nothing gets copied via COPY_KERNEL_PARAMETERS because there are no `console=` parameters to copy, but all the SERIAL_CONSOLE_DEVICE* parameters would still be respected.

Reply:
I fixed the "provided USE_SERIAL_CONSOLE is not set to 'no'"
where it was false in default.conf via
https://github.com/rear/rear/pull/2961/commits/478355980e80eecab90fe586f2ce7a49738c9839